experimental/Schemes/Sheaves.jl

```diff
@@ -0,0 +1,105 @@
+export AbsSheaf
+export space, restriction_map
+export SheafOnScheme
+export RingOfRegularFunctions
+
+########################################################################
+# The AbsSheaf interface                                               #
+########################################################################
+@Markdown.doc """
+    space(F::AbsSheaf) 
+
+For a sheaf ``ℱ`` on a space ``X`` return ``X``.
+"""
+function space(F::AbsSheaf) 
+  return space(underlying_sheaf(F))
+end
+
+@Markdown.doc """
+    (F::AbsSheaf{<:Any, OpenType})(U::T) where {T<:OpenType}
+
+For a sheaf ``ℱ`` on a space ``X`` and an (admissible) open set 
+``U ⊂ X`` check whether ``U`` is open in ``X`` and return ``ℱ(U)``.
+"""
+function (F::AbsSheaf{<:Any, OpenType, OutputType})(U::T) where {OpenType, OutputType, T<:OpenType}
+  return (underlying_sheaf(F))(U)::OutputType
+end
+
+@Markdown.doc """
+    restriction_map(F::AbsSheaf{<:Any, OpenType, OutputType},
+                    U::Type1, V::Type2
+                   ) where {
+                     OpenType, OutputType, 
+                     Type1<:OpenType, Type2<:OpenType
+                   }
+
+For a sheaf ``ℱ`` on a space ``X`` and an (admissible) pair of 
+open sets ``U, V ⊂ X`` check whether ``U ⊂ V ⊂ X`` are open and 
+return the restriction map ``ℱ(V) → ℱ(U)``.
+"""
```

Comment:
```suggestion
    restriction_map(F::AbsSheaf, U, V)
    
For a sheaf ``ℱ`` on a space ``X`` and an (admissible) pair of 
open sets ``U, V ⊂ X`` check whether ``U ⊂ V ⊂ X`` are open and 
return the restriction map ``ℱ(V) → ℱ(U)``.
"""
```
We can hide the less important parts of the signature.